docs/proposals/scheduling/20220510-load-aware-scheduling.md

```diff
@@ -0,0 +1,159 @@
+
+If the last update time of the current node's NodeMetric exceeds the configured maximum update time window, the node will get 0 points.
+
+#### Plugin configuration
+
+```go
+
+type LoadAwareSchedulingArgs struct {
+  metav1.TypeMeta
+
+  FilterUnhealthyNodeMetrics         bool                          `json:"filterUnhealthyNodeMetrics,omitempty"`
+  NodeMetricExpiredSeconds           int64                         `json:"nodeMetricExpiredSeconds,omitempty"`
+  ResourceWeights                    map[corev1.ResourceName]int64 `json:"resourceWeights,omitempty"`
+  UsageThresholds                    map[corev1.ResourceName]int64 `json:"usageThresholds,omitempty"`
+  EstimatedScalingFactors            map[corev1.ResourceName]int64 `json:"estimatedScalingFactors,omitempty"`
+}
+
+```
+
+- `FilterUnhealthyNodeMetrics` indicates whether to filter nodes where koordlet fails to update NodeMetric.
+- `NodeMetricExpiredSeconds` indicates the NodeMetric expired in seconds. When NodeMetrics expired, the node is considered abnormal.
+- `ResourceWeights` indicates the weights of resources. The weights of CPU and Memory are both 1 by default.
+- `UsageThresholds` indicates the resource utilization threshold, the default for CPU is 65%, and the default for memory is 95%.
+- `EstimatedScalingFactors` indicates the factor when estimating resource usage. The default value of CPU is 85%, and the default value of Memory is 70%.
+
+`FilterUnhealthyNodeMetrics` controls the filter behavior, if it is false, `NodeMetricExpiredSeconds` can still be used when scoring.
+
+#### Custom NodeMetric update Period
+
+This plugin is dependent on NodeMetric's reporting period. Different reporting periods need to be set according to different scenarios and workloads. If the reporting period is relatively long, koordlet needs to aggregate within the reporting period to ensure the effect of the metrics. Therefore, NodeMetricSpec needs to be extended to support user-defined reporting period and aggregation period. Users can modify `slo-controller-config` to complete the corresponding configuration, and the controller in `koord-manager` will be responsible for updating the reporting period and aggregation period fields of NodeMetrics of related nodes.
+
+```go
+type NodeMetricSpec struct {
+  CollectPolicy *MetricCollectPolicy `json:"metricCollectPolicy,omitempty"`
+}
+
+type MetricCollectPolicy struct {
+  AggregateDurationSeconds `json:"aggregateDurationSeconds,omitempty"`
```

Comment:
How about
```
AggregateDuration time.Duration
ReportPeriod      time.Duration
```
WDYT @eahydra

Reply:
Seconds suffix is recommend by K8s API community